Orange/widgets/unsupervised/owmanifoldlearning.py

```diff
@@ -93,11 +93,29 @@ class TSNEParametersEditor(ManifoldParametersEditor):
     # rename l2 to Euclidean
     metric_values = [("l2", "Euclidean")] + metric_values
 
+    perplexity = Setting(30)
+    early_exaggeration = Setting(4)
+    learning_rate = Setting(1000)
+    n_iter = Setting(1000)
+
+    init_index = Setting(0)
+    init_values = [("random", "Random"), ("pca", "PCA")]
+
     def __init__(self, parent):
         super().__init__(parent)
+
         self.metric_combo = self._create_combo_parameter(
             "metric", "Metric:")
-        self.parameters["init"] = "pca"
+        self.perplexity_spin = self._create_spin_parameter(
+            "perplexity", 0, 1000, "Perplexity:")
```

Comment:
I am not sure about this one, but sklearn doc says to consider values 5-50. I think spin limits of 1-100 should be fine?